src/overlays/actors/ovl_Bg_Icicle/z_bg_icicle.c

```diff
@@ -21,26 +52,199 @@ const ActorInit Bg_Icicle_InitVars = {
+    Collider_UpdateCylinder(thisx, &this->collider);
+
+    paramsHigh = (thisx->params >> 8) & 0xFF;
+    paramsMid = (thisx->params >> 2) & 0x3F;
+    this->unk_161 = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 3;
+
+    if (thisx->params == ICICLE_STALAGMITE_RANDOM_DROP || thisx->params == ICICLE_STALAGMITE_FIXED_DROP) {
+        this->unk_160 = ((thisx->params == ICICLE_STALAGMITE_RANDOM_DROP) ? paramsHigh : paramsMid);
+        this->actionFunc = BgIcicle_DoNothing;
+    } else {
+        this->dyna.actor.shape.rot.x = -0x8000;
+        this->dyna.actor.shape.yOffset = 1200.0f;
+        this->actionFunc = BgIcicle_Wait;
+    }
+}
+
+void BgIcicle_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIcicle* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgIcicle_Break(BgIcicle* this, GlobalContext* globalCtx, f32 arg2) {
+    static Vec3f accel = { 0.0f, -1.0f, 0.0f };
+    static Color_RGBA8 primColor = { 170, 255, 255, 255 };
+    static Color_RGBA8 envColor = { 0, 50, 100, 255 };
+    Vec3f velocity;
+    Vec3f pos;
+    s32 j;
+    s32 i;
+
+    func_800F0568(globalCtx, &this->dyna.actor.world.pos, 30, 0x28CB);
+
+    for (i = 0; i < 2; i++) {
+        for (j = 0; j < 10; j++) {
+            pos.x = this->dyna.actor.world.pos.x + randPlusMinusPoint5Scaled(8.0f);
+            pos.y = this->dyna.actor.world.pos.y + (Rand_ZeroOne() * arg2) + (i * arg2);
+            pos.z = this->dyna.actor.world.pos.z + randPlusMinusPoint5Scaled(8.0f);
+
+            velocity.x = randPlusMinusPoint5Scaled(7.0f);
+            velocity.z = randPlusMinusPoint5Scaled(7.0f);
+            velocity.y = (Rand_ZeroOne() * 4.0f) + 8.0f;
+
+            EffectSsEnIce_Spawn(globalCtx, &pos, (Rand_ZeroOne() * 0.2f) + 0.1f, &velocity, &accel, &primColor,
+                                &envColor, 30);
+        }
+    }
+}
+
+void BgIcicle_DoNothing(BgIcicle* this, GlobalContext* globalCtx) {
+}
+
+void BgIcicle_Wait(BgIcicle* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.xzDistToPlayer < 60.0f) {
+        this->shiverTimer = 10;
+        this->actionFunc = BgIcicle_Shiver;
+    }
+}
+
+void BgIcicle_Shiver(BgIcicle* this, GlobalContext* globalCtx) {
+    s32 randSign;
+    f32 rand;
+
+    if (this->shiverTimer != 0) {
+        this->shiverTimer--;
+    }
+
+    if (!(this->shiverTimer % 4)) {
+        Audio_PlayActorSound2(this, 0x28D4);
+    }
+
+    if (this->shiverTimer == 0) {
+        this->dyna.actor.world.pos.x = this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z;
+
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->actionFunc = BgIcicle_Fall;
+    } else {
+        rand = Rand_ZeroOne();
+        randSign = (Rand_ZeroOne() < 0.5f ? -1 : 1);
```

Comment:
```suggestion
        randSign = Rand_ZeroOne() < 0.5f ? -1 : 1;
```

Reply:
I like brackets round my ternaries: makes them easier to read IMO.